firedrake/preconditioners/asm.py

```diff
@@ -229,18 +227,20 @@ def get_patches(self, V):
 
             # Create point list from mesh DM
             star, _ = mesh_dm.getTransitiveClosure(seed, useCone=False)
-            pt_array = set()
-            for pt in star.tolist():
+            star = order_points(mesh_dm, star, ordering, self.prefix)
+            pt_array = []
+            for pt in reversed(star):
                 closure, _ = mesh_dm.getTransitiveClosure(pt, useCone=True)
-                pt_array.update(closure.tolist())
+                pt_array.extend(closure)
+            # Grab unique points with stable ordering
+            pt_array = list(reversed(dict.fromkeys(pt_array)))
 
-            pt_array = order_points(mesh_dm, list(pt_array), ordering, self.prefix)
             # Get DoF indices for patch
             indices = []
             for (i, W) in enumerate(V):
                 section = W.dm.getDefaultSection()
                 if i in exclude_subspaces:
-                    loop_list = [seed]
+                    loop_list = star
```

Comment:
I agree that `star` is generally better than `[seed]` here, but I worry that this is actually a bigger change than is needed for this PR.  I also see reasons for having both options available.  Can I propose that we revert this change, and offer to address this in a separate PR (adding `exclude_subspace_type` or similar to specify behaviour)?

Reply:
For high order Taylor-Hood, excluding subspaces like this and not applying any other relaxation on the pressure would be catastrophic, the preconditioner would be singular. Is the common thing to combine this with another relaxation on the pressure?

Reply:
I completely agree that it would be catastrophic.  You can, however, stitch together either single-pressure-dof patches (for P2/Q2) or vertex/edge/face/element pressure patches in a composite preconditioner for Taylor-Hood, and it works just fine.  I would agree that `star` should become the default here, but I'd like to have both families be accessible.

Reply:
Are you suggesting Vanka patches on V*Q constructed around vertices and only including the vertex DOF of Q + Edge-star patches on V*Q including edge, faces, and interiors of both V and Q? This sounds like a bit more expensive than Vanka patches on V*Q where we include the DOFs of Q from the star (but not it's closure).